test/python/circuit/library/test_mcmt.py

```diff
@@ -285,6 +285,19 @@ def test_gate_with_parameters_vchain(self):
         self.assertEqual(circuit.num_parameters, 1)
         self.assertIs(circuit.parameters[0], theta)
 
+    def test_mcmt_circuit_as_gate(self):
+        """Test the MCMT plugin is only triggered for the gate, not the same-named circuit.
+
+        Regression test of #13563.
+        """
+        circuit = QuantumCircuit(2)
+        gate = RYGate(0.1)
+        mcmt = MCMT(gate=gate, num_ctrl_qubits=1, num_target_qubits=1)
+        circuit.append(mcmt, circuit.qubits)  # append the MCMT circuit as gate called "MCMT"
```

Comment:
A bit off-topic. Here we are appending the _quantum circuit_ ``mcmt``  to the quantum circuit ``circuit``, yet the docstring for ``append`` states ``instruction: Operation | CircuitInstruction``. The code does work, since a ``QuantumCircuit`` defines a ``to_instruction`` method, which ``append`` tries to use when the instruction is not of type ``Operation``. Would it be slightly cleaner to change to ``circuit.append(mcmt.to_instruction(), ...)``, or should we change the typehint for the ``append`` method?